lib/esm/connection/connection_manager.rb

```diff
@@ -44,6 +53,21 @@ def check_lobby
 
         client.close
       end
+
+      # Traverse the connections and ping them every so often to determine if they are
+      # still connected or not by sending them a heartbeat request
+      def check_connections
+        id = @ids_to_check.shift
+        return if id.nil?
+
+        client = find(id)
+        return if client.nil?
+
+        response = client.write(type: :heartbeat).wait_for_response(@heartbeat_timeout)
+        return @ids_to_check << id if response.fulfilled?
+
+        client.close
```

Comment:
As far as I can tell this is only called in overseer.rb which doesn't contain a notification if/when the connection is found to be disconnected. Would it be prudent to add some sort of output (either log with timestamp/notice or user ping) when client is closed due to failed connection?

Reply:
Good question, and in this circumstance yes. However, the code isn't in this file because it is more closely related to the client itself than the connection manager. The  [`ESM::Connection::Client#on_disconnect`](https://github.com/itsthedevman/esm_bot/pull/21/files#diff-ff67eca518419b77f2e0cc06fbffb69a8d26d0a502e39be5e37a2f6ff0176b93R133-R143) method handles all of that